client/src/Try/Container.purs

```diff
@@ -442,6 +438,36 @@ renderCompilerErrors errors = do
     , renderPlaintext message
     ]
 
+shareButton :: forall q i o. H.Component q i o Aff
+shareButton =  H.mkComponent
+  { initialState: \_ -> 0
+  , render
+  , eval: H.mkEval $ H.defaultEval
+      { handleAction = handleAction
+      }
+  }
+  where 
+  handleAction :: Unit -> H.HalogenM Int Unit () o Aff Unit
+  handleAction _ = do
+      url <- H.liftEffect $ window >>= location >>= href
+      H.liftAff $ makeAff \f -> do
+        runEffectFn3 copyToClipboard url (f (Right unit)) (f (Left $ Aff.error "Failed to copy to clipboard"))
+        mempty
+      H.modify_ (_ + 1)
+      H.liftAff $ delay (1_500.0 # Milliseconds) 
+      H.modify_ (_ - 1)
```

Comment:
Maybe this state would be more clear if it were a `showConfirmation :: Boolean` instead of `Int`? I used `Int` so that repeated clicks would keep the confirmation message showing, but that's probably not necessary.

Reply:
Doesn't this block the thread because you're not wrapping the `delay` in an `H.fork`? And shouldn't some debouncer be used here:
1. user clicks the button, which copies the content to the clipboard and starts a debouncer. If the copy doesn't fail, a "Copied to Clipboard" message is displayed. If it does fail, a "Failed to Copy" is displayed.
2. if user clicks again before debouncer ends, debounder restarts
3. if debouncer times out, then the text changes back to "Share URL"

Then, the state would be just `showConfirmation :: Boolean` or perhaps `Maybe String` where `Nothing = "Share URL"` and `Just s = s` where `s` is either "Failed to Copy" or "Copied to Clipboard". If you initialized this component with a `Ref`, you could use that to store the reference to the debouncer's ref.

Reply:
Wouldn't debouncing mean something like: _we delay writing to clipboard until **N** milliseconds have passed without another click event_? I think I'd still want to write to clipboard immediately, but just keep the notification message showing for some time after. I'm new to Halogen, so there might be a way to leverage debouncing to accomplish that that I'm unaware of; maybe I'd need to setup an event emitter and then debounce some "hide notification message" event?
I've got a prototyped solution using `H.fork` and storing the `ForkId` in state here: https://try.purescript.org/?gist=5605231047155a3bf4b834daa03b510e
I'm not sure if there's a simpler way, but that will at least allow showing either a success or failure message as appropriate. I'll plan on going with that unless I find a simpler solution.

Reply:
Using `H.fork` (and cancelling previous `handleAction`) here 6eb73be18a074c3f82065eabda4e450e981e77a7

Reply:
> Wouldn't debouncing mean something like: we delay writing to clipboard until N milliseconds have passed without another click event?

No, we'd write to the clipboard immediately. We would _delay_ when we change from "Copied to Clipboard" back to the original text of "Share Link". If multiple clicks happen, we keep displaying "Copied to Clipboard". Once the debouncer times out because the user hasn't clicked that button for a while, we change back to "Share Link".

Reply:
I think the recent changes implement that behavior, though I'm not explicitly using a `debounce` function.